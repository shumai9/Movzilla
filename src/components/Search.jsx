import React, {Component} from 'react';
import Suggestions from './Suggestions';
import '../style/Search.css';

const Api_key = `${process.env.REACT_APP_OMDb_API_KEY}`;
const API_URL = 'http://www.omdbapi.com/?'

class Search extends Component {
  constructor (props){
    super(props);
    this.state = {
      query: '',
      search_results: [],
      loaded: false
    }
  }
  getData=()=> {
    fetch(`${API_URL}s=${this.state.query}&type=movie&apikey=${Api_key}`)
    .then(res => res.json())
    .then(result => {
      if(result.Response){
        this.setState({
          search_results: result.Search,
          count: result.totalResults,
          loaded: true
        })
      }
    })
    .catch(
      e => {console.log("From movie", e);}
    );
  }
 
  handleSearchInput = (e) => {
    e.preventDefault();
    this.setState(
      { query: this.search.value },
      () => { 
        if(this.state.query && this.state.query.length > 1){
          if(this.state.query !== ''){
            this.getData();
          }
        } 
      }
    )
  }
  render() {
    const results = this.state.search_results;
    return (
      <div>
       <form onSubmit={this.handleSearchInput}>
        <input 
          className="search title"
          onChange={this.handleSearchInput}
          onFocus={this.clearSearchInput}
          type="text"
          ref={input => {this.search = input}}          
          placeholder="type movie title here"
        />
       <Suggestions results={results}/>
        </form>
        <p>Total search results: {this.state.count}</p>        
      </div>
    )
  }
}


export default Search;