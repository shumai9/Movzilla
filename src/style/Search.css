 form {
    width: auto;
    position: absolute;
    top: 10.5rem; 
    right: 1rem;
 }
 
 .search, .suggestions{
    width: 20rem;
    
    
}