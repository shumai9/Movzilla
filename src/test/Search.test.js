import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from '../components/Search';


it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Search />, div);
	ReactDOM.unmountComponentAtNode(div);
  });
  
it('renders correctly', () =>{
const tree = renderer
  .create(<Search />)
  .toJSON();
expect(tree).toMatchSnapshot();
}); 
