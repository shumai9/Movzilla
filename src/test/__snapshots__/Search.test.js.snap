// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="search title"
      onChange={[Function]}
      placeholder="type movie title here"
      type="text"
    />
    <ul
      className="suggestions"
    />
  </form>
  <p>
    Total search results: 
  </p>
</div>
`;
